*** Makefile	2014-02-18 18:45:52.776442147 +0100
--- Makefile	2014-02-18 18:46:53.774458734 +0100
***************
*** 70,76 ****
  #
  
  # set the base of the Installation 
! # BASEDIR = /mnt
  
  # path to the net-lib support library. Default: lib
  NET_LIB_PATH = lib
--- 70,76 ----
  #
  
  # set the base of the Installation 
! BASEDIR = /tmp/pkg
  
  # path to the net-lib support library. Default: lib
  NET_LIB_PATH = lib
*** lib/inet_sr.c	2014-02-18 19:07:47.699746388 +0100
--- lib/inet_sr.c	2014-02-18 19:06:38.542732917 +0100
***************
*** 104,110 ****
         isnet = 1; break;
      case 2:
         isnet = 0; break;
!     default:
      }
  
      /* Fill in the other fields. */
--- 104,110 ----
         isnet = 1; break;
      case 2:
         isnet = 0; break;
!     default: break;
      }
  
      /* Fill in the other fields. */
*** hostname.c	2014-02-18 19:11:33.167788590 +0100
--- hostname.c	2014-02-18 19:12:16.955796490 +0100
***************
*** 77,83 ****
          case EINVAL:
              fprintf(stderr, _("%s: name too long\n"), program_name);
              break;
!         default:
          }
  	exit(1);
      }
--- 77,83 ----
          case EINVAL:
              fprintf(stderr, _("%s: name too long\n"), program_name);
              break;
!         default: break;
          }
  	exit(1);
      }
***************
*** 97,103 ****
  	case EINVAL:
  	    fprintf(stderr, _("%s: name too long\n"), program_name);
  	    break;
! 	default:
  	}
  	exit(1);
      };
--- 97,103 ----
  	case EINVAL:
  	    fprintf(stderr, _("%s: name too long\n"), program_name);
  	    break;
! 	default: break;
  	}
  	exit(1);
      };
***************
*** 116,122 ****
  	case EINVAL:
  	    fprintf(stderr, _("%s: name too long\n"), program_name);
  	    break;
! 	default:
  	}
  	exit(1);
      };
--- 116,122 ----
  	case EINVAL:
  	    fprintf(stderr, _("%s: name too long\n"), program_name);
  	    break;
! 	default: break;
  	}
  	exit(1);
      };
***************
*** 173,179 ****
  	    *p = '\0';
  	printf("%s\n", hp->h_name);
  	break;
!     default:
      }
  }
  
--- 173,179 ----
  	    *p = '\0';
  	printf("%s\n", hp->h_name);
  	break;
!     default: break;
      }
  }
  
